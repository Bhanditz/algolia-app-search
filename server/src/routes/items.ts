import { Context } from 'koa';
import { removeItem, addItem } from '../services/algolia';

export const deleteRoute = async (ctx: Context) => {
  const { id } = ctx.params;

  if (id == null) {
    throw new Error('Please provide an id');
  }

  const item = await removeItem(id);

  ctx.body = { status: 'success', item };
};

export const createRoute = async (ctx: Context) => {
  const { item } = ctx.params;

  if (item == null) {
    throw new Error('Please provide an item');
  }

  await addItem(item);

  ctx.body = { status: 'success', item };
};
