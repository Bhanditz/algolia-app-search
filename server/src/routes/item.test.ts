import { createRoute, deleteRoute } from './items';

jest.mock('../services/algolia', () => ({
  addItem: (item: Item) => {
    return item;
  },
  removeItem: (id: string) => {
    return { id: 'item' };
  },
}));

describe('routes', () => {
  describe('items', () => {
    it('can create an item', async () => {
      const item = {
        category: 'Games',
        rating: 4.5,
        name: 'Draw Something Free',
        image:
          'http://a567.phobos.apple.com/us/r1000/112/Purple/v4/f2/ec/56/f2ec5676-3c96-ab7e-39f2-5e98e9662040/Icon.png',
        link: 'https://itunes.apple.com/us/app/draw-something-free/id488628250',
        ratingCount: 1670278,
        price: '0 USD',
      };
      const context = {
        params: {
          item,
        },
      };

      const result = await createRoute(context);

      expect(context.body).toEqual({ status: 'success', item });
    });

    it('does check parameters on create', async () => {
      return createRoute({ params: {} }).catch(e => {
        expect(e).toMatchSnapshot();
      });
    });

    it('can remove an item', async () => {
      const context = {
        params: {
          id: 'test',
        },
      };

      const result = await deleteRoute(context);

      expect(context.body).toEqual({ status: 'success', item: { id: 'item' } });
    });

    it('does check parameters on remove', async () => {
      return deleteRoute({ params: {} }).catch(e => {
        expect(e).toMatchSnapshot();
      });
    });
  });
});
