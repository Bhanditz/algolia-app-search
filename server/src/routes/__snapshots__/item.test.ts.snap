// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`routes items can create an item 1`] = `
Object {
  "info": Object {
    "category": "Games",
    "image": "http://a567.phobos.apple.com/us/r1000/112/Purple/v4/f2/ec/56/f2ec5676-3c96-ab7e-39f2-5e98e9662040/Icon.png",
    "link": "https://itunes.apple.com/us/app/draw-something-free/id488628250",
    "name": "Draw Something Free",
    "price": "0 USD",
    "rating": 4.5,
    "ratingCount": 1670278,
  },
  "item": Object {
    "category": "Games",
    "image": "http://a567.phobos.apple.com/us/r1000/112/Purple/v4/f2/ec/56/f2ec5676-3c96-ab7e-39f2-5e98e9662040/Icon.png",
    "link": "https://itunes.apple.com/us/app/draw-something-free/id488628250",
    "name": "Draw Something Free",
    "price": "0 USD",
    "rating": 4.5,
    "ratingCount": 1670278,
  },
  "status": "success",
}
`;

exports[`routes items does check parameters on create 1`] = `[Error: No entity is provided]`;

exports[`routes items does check parameters on remove 1`] = `[Error: Please provide an id]`;
